crates/runtime/src/logic.rs

```diff
@@ -335,6 +336,26 @@ impl VMHostFunctions<'_> {
         Ok(())
     }
 
+    /// Sends an action to the host.
+    ///
+    /// After a storage event, other nodes need to be updated. Consequently, the
+    /// host must be informed about the action that was taken. This function
+    /// sends that action to the host, which can then be used to update the
+    /// network.
+    ///
+    pub fn send_action(&mut self, action_ptr: u64, action_len: u64) -> VMLogicResult<()> {
+        let action_bytes = self.read_guest_memory(action_ptr, action_len)?;
+        let action: Action = BorshDeserialize::deserialize(&mut &action_bytes[..])
+            .map_err(|_err| HostError::DeserializationError)?;
+
+        println!("Action received: {:?}", action);
+
+        // TODO: What should this now call in order to propagate the Action to other
+        // TODO: nodes in the network? We need to get it into Node::push_action()
```

Comment:
@miraclx

Reply:
The rest of the flow would work the way `value_return` / events / logs work.

For example:

https://github.com/calimero-network/core/blob/13cfbd431eeddb4542bb08e32953ac88b408b41f/crates/runtime/src/logic.rs#L302

and eventually, makes it to the `Outcome`

https://github.com/calimero-network/core/blob/13cfbd431eeddb4542bb08e32953ac88b408b41f/crates/runtime/src/logic.rs#L169-L173